Bool1DArray[] data_2d